materials: Covert to es6 class.

linbingquan opened the pull request.

raub committed:
fixed missing space in GLBufferAttribute example

raub committed:
fixed info title in GLBufferAttribute example

raub committed:
Merge branch 'master' of https://github.com/mrdoob/three.js into dev

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Add culling warnings for GLBufferAttribute, update the example

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Fix GLBufferAttribute example

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Add  parameter to the constructor

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Update GLBufferAttribute

raub committed:
Merge branch 'master' of https://github.com/mrdoob/three.js into dev

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Add example screenshot

raub committed:
Fix e2e test

raub committed:
Fix docs list entry

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Add zh doc entry

WestLangley committed:
Set sRGB encoding. Also some clean up.

WestLangley committed:
Disable stencil buffer by default

ianpurvis committed:
AudioListener: Convert to es6 class

ianpurvis committed:
PositionalAudio: Convert to es6 class

ianpurvis committed:
Audio: Convert to es6 class

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

raub committed:
Update GLBufferAttribute

raub committed:
Update screenshot for GLBufferAttribute example

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

yomotsu committed:
Vector2

yomotsu committed:
missing semi

yomotsu committed:
Vector3, Vector4

yomotsu committed:
solve hoisting in Vector3

yomotsu committed:
Matrix3, Matrix4

fraguada committed:
adding incomplete loader work

yomotsu committed:
Quaternion

yomotsu committed:
Euler

yomotsu committed:
fix static value of Euler

linbingquan committed:
materials: Covert to es6 class.

fraguada committed:
Some cleanup of copied code

mrdoob committed:
Merge pull request #20003 from ianpurvis/audio-es6-classes

Audio: Convert audio/ to es6 classes

mrdoob committed:
Updated builds.

Methuselah96 committed:
Remove outdated examples of ArrowHelper

DefinitelyMaybe commented:
@linbingquan calling out what you're working on helps the rest of us coordinate with you, instead of doubling up on some of the work

yomotsu committed:
move internal use instances to the end of the file

J-Rojas committed:
Fix for #19959. Add 'glslVersion' property to ShaderMaterial and prepend the version string when it's value is non-null.

J-Rojas committed:
Fix linting errors

DefinitelyMaybe committed:
src/scenes: move to es6 classes

fraguada committed:
Ahhhh! It works!!!

fraguada committed:
Check that the buffer is actually the 3dm file.

DavidPeicho committed:
program: make unroll loop regexp accept other coding style

linbingquan committed:
Updated: Clean up.

Methuselah96 committed:
Remove from Chinese docs as well

fraguada committed:
getting meshes out!

fraguada committed:
cleanup

fraguada committed:
cleanup console

FishOrBear committed:
Update MeshPhysicalMaterial.d.ts

J-Rojas committed:
Conditionally apply automatic GLSL 3.0 conversions.
Add GLSL1 and GLSL3 version constants

J-Rojas committed:
Add GLSL1 and GLSL3 declarations for Typescript

J-Rojas committed:
Fix linting errors

fraguada committed:
added attributes and materials

Commit:
Added support for tracks using cubic spline interpolation to AnimationUtils.makeClipAdditive

J-Rojas committed:
Remove #version from GLSL constants... only the version string should be used.

Mugen87 committed:
ParametricGeometry: Revert class change.

J-Rojas committed:
Address PR feedback

Commit:
Fixed index lookup for non-quaternions in AnimationUtils.makeClipAdditive and in PropertyMixer._setAdditiveIdentityQuaternion

fraguada committed:
Merge pull request #1 from fraguada/fraguada-Rhino3dmLoader

Fraguada rhino3dm loader

Mugen87 committed:
Examples: Update chevrotain module.

JohannesDeml committed:
Fix orthographic camera rendering for CSS3D on safari on High DPI displays

JohannesDeml committed:
Fix css perspective camera for safari when looking straight forward

fraguada committed:
Updated decode code and added some material props

JohannesDeml committed:
Change remove perspective camera "fix"

This results in wrong rendering for normal displays

JohannesDeml committed:
Change match to test for safari check

Co-authored-by: Akihiro Oyamada <admin@yomotsu.net>

DefinitelyMaybe committed:
perfer const

DefinitelyMaybe committed:
adding clarity.

DefinitelyMaybe committed:
following formatting.

fraguada committed:
initial parsing of layers and views

fraguada committed:
Added extrusion support, groups, settings

fraguada committed:
cleanup.

dmnsgn committed:
Add @default values to TypeScript declaration files

mrdoob committed:
Merge pull request #20031 from DefinitelyMaybe/prefer-const

Package.json: Added prefer-const

mrdoob committed:
Merge pull request #20027 from Mugen87/dev51

Examples: Update chevrotain module.

mrdoob committed:
Merge pull request #20023 from Mugen87/dev46

ParametricGeometry: Revert class change.

mrdoob committed:
Merge pull request #20017 from FishOrBear/patch-5

Update MeshPhysicalMaterial.d.ts

mrdoob committed:
Merge pull request #20004 from Methuselah96/patch-1

Docs: Remove outdated examples of ArrowHelper

mrdoob committed:
Merge pull request #20015 from linbingquan/dev-cleanup-examples

Examples: Clean up.

mrdoob committed:
Updated builds.

fraguada committed:
adding layer visibility toggle to example.

fraguada committed:
fix example layer visibility functionality

mrdoob committed:
Make equirectangular reflections / refractions backwards compatible. (#19911)

* Remove Equirect Reflection/Refraction Mapping

* Reverted EquirectangularReflectionMapping and EquirectangularRefractionMapping removals.

* Added EquirectangularTexture and EquirectangularTextureLoader.

* WebGLCubeRenderTarget: Avoid using renderer.render() in fromEquirectangularTexture().

* WebGLBackground: Support EquirectangularTextures as background.

* Examples: Simplified webgl_materials_envmaps example.

* Fixed EquirectangularTextureLoader.d.ts

* Removed EquirectangularTextureLoader

* WebGLBackground: Fixed breaking when using Color as background.

* Examples: Fixed FBXLoader and LWOLoader.

* WebGLTextures: Moved cubeTexture code to its own class.

* WebGLCubeMaps: Handle CubeUVReflectionMapping and CubeUVRefractionMapping.

* WebGLCubeRenderTarget: Renamed isWebGLCubeTexture hack to isWebGLCubeRenderTargetTexture.

* Removed EquirectangularTexture

* WebGLCubeMaps: Simplified and fixed some breakage.

* Reverted more removals.

* WebGLRenderer: Check object.geometry before updating attributes.

* WebGLRenderer: Added getRenderList and setRenderList.

* WebGLBackground: Added WebGLCubeRenderTarget support back.

* Reverted comment removal.

* WebGLCubeRenderTarget: Avoid using mipmaps in fromEquirectangularTexture.

* WebGLCubeRenderTarget: Reverted generateMipmaps/migFilter/magFilter removal.

* WebGLCubeMaps: Added DataTexture support.

* Examples: Added more detail to sphere in webgl_materials_envmap.

* WebGLCubeRenderTarget: Improved LinearMipmapLinearFilter handling.

* WebGLCubeMaps: Added support for CompressedTextures.

Co-authored-by: WestLangley <WestLangley@users.noreply.github.com>

mrdoob committed:
Updated builds.

DefinitelyMaybe committed:
Package.json: Added "lint-fix" script (#20033)

* add --fix parameter for eslint

* removing --fix comment

* new script

mrdoob committed:
Merge pull request #20032 from DefinitelyMaybe/package.json

Package.json: Adding clarity to build-closure script

mrdoob committed:
Merge pull request #20028 from JohannesDeml/css3d-rendering-issues-safari

Fix orthographic camera rendering for CSS3D on safari on High DPI displays

mrdoob committed:
CSS3DRenderer: Clean up.

yomotsu committed:
Euler.DefaultOrder should be writable

encadyma committed:
Fix Euler property typo in setFromEuler

euler.order -> euler._order

Mugen87 committed:
WebGLCubeMaps: Fix runtime error when image is undefined.

Mugen87 committed:
TS: Add WebGLCubeMaps.

Mugen87 committed:
Clean up.

fraguada committed:
Added support for curves and textdots.

Started instance definitions / references

mrdoob committed:
Merge pull request #20043 from Mugen87/dev51

WebGLCubeMaps: Fix runtime error when image is undefined.

mrdoob committed:
Merge pull request #20044 from Mugen87/dev46

TS: Add WebGLCubeMaps.

mrdoob committed:
Merge pull request #20042 from encadyma/kmo-patch-euler

Fixed Euler property typo in setFromEuler

shawn0326 committed:
Add index cache for vao

fraguada committed:
refactor and code styling

fraguada committed:
Handling Instances, needs cleanup

fraguada committed:
Code Style cleanup

mrdoob committed:
Updated builds.

Mugen87 committed:
WebGLRenderer: Fix equirect texture conversion.

yomotsu committed:
revert Euler.RotationOrders to be writable

fraguada committed:
cleanup, adding screenshot

reverting unintended change to draco loader

Sammius committed:
Update MaterialLoader.js

Typo

fraguada committed:
updated sample model, added d.ts, screenshot

fraguada committed:
Cleanup formating.

fraguada committed:
adding request header

mrdoob committed:
Merge pull request #20050 from Sammius/patch-1

MaterialLoader: Add missing break statement in parse().

Commit:
GLTFLoader: Fix regression unecessarily choosing MeshPhysicalMaterial.

greggman committed:
check if cachedAttribute is undefined

fraguada committed:
deleting unused model

mrdoob committed:
Merge pull request #20047 from mcneel/3DMLoader

Added Rhino3dmLoader

mrdoob committed:
Merge pull request #20056 from greggman/check-cached-attribute

WebGLBindingStates: check if cachedAttribute is undefined

mrdoob committed:
Merge pull request #20045 from shawn0326/fix-vao

WebGLBindingStates: Add index to cache.

mrdoob committed:
Merge pull request #20053 from donmccurdy/bug-gltfloader-getmaterialtype

GLTFLoader: Fix regression unnecessarily choosing MeshPhysicalMaterial.

linbingquan committed:
Updated: webgl_loader_3dm screenshot.

mrdoob committed:
Merge pull request #20063 from linbingquan/dev-screenshot

Updated: webgl_loader_3dm screenshot.

linbingquan committed:
Updated: package.json

mrdoob committed:
Merge pull request #20064 from linbingquan/dev-packagejson

Updated: package.json

WestLangley committed:
Revert Fix orthographic camera rendering for CSS3D on safari on High DPI displays

WestLangley committed:
Added onWindowResize callback

mrdoob committed:
Merge pull request #20067 from WestLangley/dev_css3d_revert

Revert Fix orthographic camera rendering for CSS3D on safari on High …

mrdoob committed:
Merge pull request #20068 from WestLangley/dev_css3d_examples

Examples: Added onWindowResize callback to CSS3D examples

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #20036 from dmnsgn/dev

Add @default values to TypeScript declaration files

yomotsu committed:
is*** should not be a getter function

takahirox committed:
Initialize Audio.source as null

mrdoob committed:
Merge pull request #20073 from takahirox/AudioSource

Initialize Audio.source as null

mrdoob committed:
Merge pull request #19997 from yomotsu/move-to-es6-classes/math

math: move to es6 classes

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #20021 from c-morten/Additive_Animation

Added support for tracks with cubic spline interpolation to AnimationUtils.makeClipAdditive

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #19800 from WestLangley/dev_examples_cubemap

Examples: Properly specify texture sRGB encoding

hassanrwx committed:
initiate: create-a-scene doc page

hassanrwx committed:
update: add Arabic lang support

hassanrwx committed:
update: add Arabic doc mapping

hassanrwx committed:
update: fix indentation issue

mrdoob committed:
Merge pull request #20075 from hassanMuhamad/dev

Docs: arabic language

mrdoob committed:
Merge pull request #20048 from Mugen87/dev46

WebGLRenderer: Fix equirect texture conversion.

mrdoob committed:
Updated builds.

raub committed:
Merge branch 'dev' of https://github.com/mrdoob/three.js into dev

DefinitelyMaybe committed:
is* adjustment

yomotsu committed:
Box2: Convert to es6 class

yomotsu committed:
Box3: Convert to es6 class

yomotsu committed:
Cylindrical.js: Convert to es6 class

yomotsu committed:
Frustum: Convert to es6 class

yomotsu committed:
Line3: Convert to es6 class

yomotsu committed:
Plane: Convert to es6 class

yomotsu committed:
Sphere: Convert to es6 class

yomotsu committed:
SphericalHarmonics3: Convert to es6 class

yomotsu committed:
Triangle: Convert to es6 class

DefinitelyMaybe commented on examples/jsm/lines/LineMaterial.js:
```diff
@@ -243,160 +243,162 @@ ShaderLib[ 'line' ] = {
+		this.dashed = false;
 
-		color: {
+		this.isLineMaterial = true;
```
[Have a quick look over this way.](https://github.com/mrdoob/three.js/issues/19986#issuecomment-674046275)

WestLangley committed:
Merge branch 'dev' of git://github.com/mrdoob/three.js into dev_rt_stencil_default

WestLangley committed:
Disable stencil buffer by default

Mugen87 committed:
Clean up.

mrdoob committed:
Examples: Improved volume example compatibility.

Mugen87 committed:
WebGLRenderer: Make uniforms handling of RectAreaLight more consistent.

mrdoob committed:
Merge pull request #20078 from Mugen87/dev46

WebGLRenderer: Make uniforms handling of RectAreaLight more consistent.

mrdoob committed:
Merge pull request #20077 from Mugen87/dev45

Clean up.

mrdoob committed:
Merge pull request #20076 from yomotsu/move-to-es6-classes/math-area-and-space

math area and space: move to es6 classes

WestLangley committed:
Disable stencil buffer by default

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #20007 from DefinitelyMaybe/src/scene--move-to-es6-classes

src/scenes: move to es6 classes

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #19926 from WestLangley/dev_rt_stencil_default

WebGLRenderTarget: Disable stencil buffer by default

mrdoob committed:
Updated builds.

hassanrwx committed:
add: browser-support doc page

hassanrwx committed:
add: installation doc page

hassanrwx committed:
fix: indentation

hassanrwx committed:
add: how-to-run-things-locally doc page

hassanrwx committed:
add: WebGL compatibility check doc page

hassanrwx committed:
add: Typescript setup doc page

hassanrwx committed:
update: documentation list mapping

mrdoob committed:
Clean up.

mrdoob committed:
Merge pull request #20079 from hassanMuhamad/dev

Docs: Added more pages

Mugen87 committed:
WebGLRenderer: Fix update of materialProperties.envMap.

mrdoob committed:
Merge pull request #20081 from Mugen87/dev46

WebGLRenderer: Fix update of materialProperties.envMap.

mrdoob committed:
Docs: Cleaned up Arabic pages.

mrdoob committed:
Updated builds.

Mugen87 committed:
WebGLClipping: Refactoring.

mrdoob committed:
Merge pull request #20084 from Mugen87/dev46

WebGLClipping: Refactoring.

Mugen87 committed:
WebGLRenderer: Avoid frame-late bug with converted env maps.

mrdoob committed:
Docs: Set navigation direction to rtl when language is ar.

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #20085 from Mugen87/dev45

WebGLRenderer: Avoid frame-late bug with converted env maps.

mrdoob committed:
Updated builds.

hassanrwx committed:
Merge pull request #1 from mrdoob/dev

Getting updates

hassanrwx committed:
fix: multiple language words arrangement

hassanrwx committed:
add: drawing-lines doc page mapping

hassanrwx committed:
add: drawing-lines Arabic doc page

hassanrwx committed:
add: creating-text doc page mapping

hassanrwx committed:
add: creating-text Arabic doc page

Mugen87 committed:
WebGLPrograms: Rename allocateBones().

yomotsu committed:
Color: Convert to es6 class

hassanrwx committed:
add: loading-3D-models doc page mapping

hassanrwx committed:
add: Loading-3D-models Arabic doc page

hassanrwx committed:
fix: direction issues

- setting (ol) element direction to rtl
- setting (code) element direction to ltr

mrdoob committed:
Merge pull request #20087 from Mugen87/dev45

WebGLPrograms: Rename allocateBones().

mrdoob committed:
Merge pull request #20088 from hassanMuhamad/dev

Docs: Added more Arabic pages

mrdoob committed:
Merge pull request #13196 from node-3d/dev

Added GLBufferAttribute

mrdoob committed:
Updated builds.

mrdoob committed:
Examples: Renamed webgl_buffergeometry_points_glbufferattribute to webgl_buffergeometry_glbufferattribute

mrdoob committed:
Merge pull request #20089 from yomotsu/move-to-es6-classes/math-color

Color: Convert to es6 class

mrdoob committed:
Updated builds.

mrdoob committed:
Merge pull request #20005 from J-Rojas/fix-19959

ShaderMaterial: Introduce glslVersion property.

Mugen87 committed:
WebGLProgram: Refactor GLSL conversion.

mrdoob committed:
Examples: Updated Ammo.js

mrdoob committed:
CannonPhysics: Clean up.

mrdoob committed:
Clean up.

mrdoob committed:
Examples: Removed redundant code from physics_cannon_instancing.

mrdoob committed:
Examples: Added AmmoPhysics.

mrdoob committed:
Examples: Fixed physics_ammo_instancing.

mrdoob committed:
Merge pull request #20090 from Mugen87/dev45

WebGLProgram: Refactor GLSL conversion.

mrdoob committed:
Updated builds.

mrdoob committed:
AmmoPhysics: Reset angular and linear velocity in setMeshPosition.

mrdoob committed:
Examples: Fixed ammo.js import.

mrdoob committed:
Examples: Removed physics_cannon_instancing.

mrdoob committed:
Merge pull request #20012 from DavidPeicho/feature/shjader-loop-unroll-improve

 WebGLProgram: make unroll loop regexp accept other coding style

mrdoob committed:
Updated builds.

Commit:
Docs: Fix broken links to transmission example.

takahirox committed:
Introduce _connected flag in Audio to fix a bug that setFilters does not
reset connection if already connected but playing is not started yet

Mugen87 committed:
ColladaExporter: Fix runtime error when processing ImageBitmap.

mrdoob committed:
Merge pull request #20092 from donmccurdy/docs-links-transmission

Docs: Fix broken links to transmission example.

mrdoob committed:
Merge pull request #20049 from Mugen87/dev51

ColladaExporter: Fix runtime error when processing ImageBitmap.

mrdoob committed:
Merge pull request #20093 from takahirox/AudioConnected

Audio: Introduce _connected flag to fix setFilters bug

Mugen87 committed:
ShaderMaterial: Improve typings and docs for glslVersion.

mrdoob committed:
Merge pull request #20095 from Mugen87/dev51

ShaderMaterial: Improve typings and docs for glslVersion.

mrdoob committed:
Editor: Removed Minify Shaders option.

Mugen87 committed:
WebGLRenderer: Remove superfluous program check in setProgram().

mrdoob committed:
Merge pull request #20097 from Mugen87/dev51

WebGLRenderer: Remove superfluous program check in setProgram().

mrdoob committed:
Updated builds.

hassanrwx committed:
add: FAQ doc page mapping

hassanrwx committed:
add: FAQ Arabic doc page

hassanrwx committed:
add: useful-links page mapping

hassanrwx committed:
add: useful-links Arabic doc page

Mugen87 committed:
Docs: Clean up.

linbingquan committed:
prototype properties: Clean up.

mrdoob committed:
Merge pull request #20099 from Mugen87/dev51

Docs: Clean up.

mrdoob committed:
Merge pull request #20098 from hassanMuhamad/dev

Finishing Manual::getting started - Arabic documentation

linbingquan committed:
Merge remote-tracking branch 'origin/dev-es6-materials' into dev
